lib/phoenix/router/route.ex

```diff
@@ -121,13 +121,16 @@ defmodule Phoenix.Router.Route do
     {_params, fwd_segments} = build_path_match(route.path)
 
     quote do
-      {Phoenix.Router.Route, {unquote(fwd_segments), unquote(route.plug), unquote(route.opts)}}
+      {
+        Phoenix.Router.Route,
+        {unquote(fwd_segments), unquote(route.plug), unquote(Macro.escape(route.opts))}
+      }
     end
   end
 
   defp build_dispatch(%Route{} = route) do
     quote do
-      {unquote(route.plug), unquote(route.opts)}
+      {unquote(route.plug), unquote(Macro.escape(route.opts))}
```

Comment:
I think this should always be an atom, because If you don't pass an atom, you will get a `** (CompileError) .../router.ex: invalid quoted expression:` when building the helpers. The error message is not very friendly, so I think we could improve it (in a different PR) with one of these options:

1. Provide a custom `ArgumentError` in the route macros (`match`, `get`, `post`, etc) is `plug_opts` is not an atom
2. Include a guard `is_atom(plug_opts)` in the route macros (`match`, `get`, `post`, etc) so it returns a `FunctionClauseError`.

I personally prefer the option 1, but I understand that it might be an edge case and the option 2 is enough.

Reply:
Forget what I said, that wouldn't work for cases like:

```elixir
action = :index
get "/hello", HelloController, action
```

The check can be done in `Phoenix.Router.Route` though.